### PR TITLE
Fix #7401: do not fail hard in Setup.hs if library interface files cannot be built

### LINKED ISSUE #7401: Cabal 3.12.1.0 install failure for `lib:Agda` - dist/build/agda/agda does not exist
When trying to upgrade `cabal-install` in Homebrew to latest 3.12.1.0 (https://github.com/Homebrew/homebrew-core/pull/178592), I saw a Cabal install failure for `lib:Agda` (which was being installed as dependency for agda2hs).

https://github.com/Homebrew/homebrew-core/actions/runs/10114758622/job/27984382041#step:3:293
```
  [429 of 429] Compiling Agda.Main        ( src/full/Agda/Main.hs, dist/build/Agda/Main.o, dist/build/Agda/Main.dyn_o )
  Installing library in /opt/homebrew/Cellar/agda/2.6.4.3_1/libexec/ghc-9.8.2-inplace/incoming/new-29976/opt/homebrew/Cellar/agda/2.6.4.3_1/libexec/ghc-9.8.2-inplace/Agd-2.6.4.3-5231c10e/lib
  Generating Agda library interface files...
  dist/build/agda/agda: readCreateProcess: posix_spawnp: does not exist (No such file or directory)
  Error: [Cabal-7125]
  Failed to build Agda-2.6.4.3 (which is required by exe:agda2hs from agda2hs-1.2). See the build log above for details.
```

---

The corresponding versions in CI were for Agda 2.6.4.3 building with Homebrew's GHC 9.8.2 on arm64 macOS Ventura. The exact same setup with Cabal 3.10.3.0 had no issue.

I was also able to reproduce issue by just running `cabal install --lib` on master branch of Agda (currently https://github.com/agda/agda/commit/5da3f3f2be5fc0d4ed6a2b828bd751e44e29292f), e.g.
```console
❯ cabal --version
cabal-install version 3.12.1.0
compiled using version 3.12.1.0 of the Cabal library

❯ cabal update
Downloading the latest package list from hackage.haskell.org
Package list of hackage.haskell.org has been updated.
The index-state is set to 2024-07-27T12:29:48Z.

❯ cabal install --lib
...
 - Agda-2.8.0 (lib:Agda) (requires build)
...
Starting     Agda-2.8.0 (all, legacy fallback: build-type is Custom)
Building     Agda-2.8.0 (all, legacy fallback: build-type is Custom)
Installing   Agda-2.8.0 (all, legacy fallback: build-type is Custom)

Failed to build Agda-2.8.0. The failure occurred during the final install
step.
...
Generating Agda library interface files...
dist/build/agda/agda: readCreateProcess: posix_spawnp: does not exist (No such file or directory)
Error: [Cabal-7125]
Failed to build Agda-2.8.0. See the build log above for details.
```

If only building but not installing lib (e.g. `cabal build` or default `exe:agda, exe:agda-mode`), there is no issue.

### PR DESCRIPTION
**Pull request by andreasabel**
- **Bump cubical to latest to include agda/cubical#1143**
- **Workaround for `cabal install --lib` (#7401)**

### CODE REVIEW

**Comment by JaSpa** [comment]
There is an unconditional `throw $ userError "FOO BAR"` left in the code:

https://github.com/agda/agda/blob/b0838414d801108e2b2907da85594e445014381f/Setup.hs#L166

**Comment by andreasabel** [comment]
Oooops!  Thanks @JaSpa for the alert!
Following Murphy's law, this affects also 2.7.0 RC2.  I'll hotfix it.
- https://github.com/agda/agda/pull/7410

File: Setup.hs [hunk 1]
```diff
@@ -130,21 +139,40 @@ generateInterfaces pd lbi = do
...
   env <- getEnvironment
-  _output <- readCreateProcess
-      (proc agda
-          [ "--interaction"
-          , "--interaction-exit-on-error"
-          , "-Werror"
-          , "-v0"
-          ])
+  handle onIOError $ do
+    throw $ userError "FOO BAR"
```
Comment (andreasabel) [inline Setup.hs:166]: Fixing this in #7410

Comment (andreasabel) [reply Setup.hs:166]: Updated 2.7.0-rc2.

**Pull request merged**